frappe/contacts/address_and_contact.py

```diff
@@ -128,25 +130,20 @@ def delete_contact_and_address(doctype, docname):
 def filter_dynamic_link_doctypes(doctype, txt, searchfield, start, page_len, filters):
 	if not txt: txt = ""
 
-	txt = txt.lower()
-	txt = "%%%s%%" % (txt)
-
-	filters.update({
-		"parent": ("like", txt)
-	})
-
 	doctypes = frappe.db.get_all("DocField", filters=filters, fields=["parent"],
 		distinct=True, as_list=True)
 
-	filters.pop("parent")
+	doctypes = tuple([d for d in doctypes if re.search(txt+".*", _(d[0]), re.IGNORECASE)])
```

Comment:
not able to understand the use case here. Can you add some tests?

Reply:
Hi @rmehta,

Thanks a lot for your review, yes I have rebased and added some test cases to make it clearer.

In two words, the use case is just that when you use ERPNext in a foreign language, you cannot use the search functionality in your language. You need to use english, which, you can imagine, is quite confusing for people and provides a very bad UX.

This is a problem only for document that are not translated in the system, i.e system documents like "DocType" and "Role" for example. There may be others but I haven't checked all yet.